gcexport.py

```diff
@@ -197,22 +198,37 @@ def sanitize_filename(name, max_length=0):
 
 
 def write_to_file(filename, content, mode, file_time=None):
-    """Helper function that persists content to file."""
-    if python3 and any((filename.endswith('.json'), filename.endswith('.gpx'), filename.endswith('.tcx'))):
-        write_file = open(filename, mode, encoding="utf-8")
-        if not isinstance(content, str):
-            content=content.decode("utf-8")
-    else:
-        write_file = open(filename, mode)
+    """
+    Helper function that persists content to a file.
+
+    :param filename:     name of the file to write
+    :param content:      content to write; with Python 2 always of type 'str',
+                         with Python 3 it can be 'bytes' or 'str'. If it's
+                         'bytes' and the mode 'w', it will be converted/decoded
+    :param mode:         'w' or 'wb'
+    :param file_time:    if given use as timestamp for the file written
+    """
+    if mode == 'w':
+        write_file = io.open(filename, mode, encoding="utf-8")
+        if isinstance(content, bytes):
+            content = content.decode("utf-8")
+    else: # 'wb'
+        write_file = io.open(filename, mode)
```

Comment:
Using `else` as assumption that it will be 'wb' can be very wrong. I would suggest a better handling of errors like this.

```suggestion
    elif mode=='wb':
        write_file = io.open(filename, mode)
    else:
        raise Exception('Unsupported file mode:' mode)
```

Reply:
I agree